### PR TITLE
Allow Symfony 5

### PR DESCRIPTION
**Pull request by kylekatarnls**

### CODE REVIEW

**Comment by tvbeek** [comment]
@kylekatarnls Do you know why we didn't get the status back from travis? And in the composer install I don't see any Symfony 5 version. Is it possible to do a test with that version?

**Comment by kylekatarnls** [comment]
I think our PHPUnit version is too old and requires some symfony/console < 5 stuff.

**Comment by kylekatarnls** [comment]
Arf, and the minimum-stability is not set so it's "stable", and Symfony 5 isn't

**Comment by kylekatarnls** [comment]
Now we have a Symfony 5 test on Travis-CI: https://travis-ci.org/pdepend/pdepend/jobs/614013760?utm_medium=notification&utm_source=github_status

Work's just starting 😈

**Comment by kylekatarnls** [comment]
Fixed.

File: src/main/php/PDepend/DependencyInjection/Configuration.php [hunk 1]
```diff
@@ -76,7 +75,7 @@ public function getConfigTreeBuilder()
         $defaultCacheDriver = ($workarounds->hasSerializeReferenceIssue()) ? 'memory' : 'file';
 
         $treeBuilder = new TreeBuilder();
-        $rootNode = $treeBuilder->root('pdepend');
+        $rootNode = $treeBuilder->getRootNode();
```
Comment (tvbeek) [inline src/main/php/PDepend/DependencyInjection/Configuration.php]: I understand the reason but doubt if it isn't more useful to create a private method that returns the treebuilder instead of creating a new class with a hard coded dependency (special in the DependencyInjection namespace 😄 )

Like:
```php
$rootNode = $this->getRootNode();

// with

private function getRootNode()
{
        $name = 'pdepend';
        if (method_exists('Symfony\\Component\\Config\\Definition\\Builder\\TreeBuilder', 'root')) {
            // Symfony < 5
            $treeBuilder = new TreeBuilder();
            return  $treeBuilder->root($name);
        }
        // Symfony 5+
        $treeBuilder = new TreeBuilder($name);
        return $treeBuilder->getRootNode();
}
```

Comment (kylekatarnls) [reply src/main/php/PDepend/DependencyInjection/Configuration.php]: We also need to return `$treeBuilder` at the end of the function so we would need this ugly syntax:

```php
private function getRootNode()
{
        $name = 'pdepend';
        if (method_exists('Symfony\\Component\\Config\\Definition\\Builder\\TreeBuilder', 'root')) {
            // Symfony < 5
            $treeBuilder = new TreeBuilder();
            return array($treeBuilder, $treeBuilder->root($name));
        }
        // Symfony 5+
        $treeBuilder = new TreeBuilder($name);
        return array($treeBuilder, $treeBuilder->getRootNode());
}
```

And a `list()` to unpack it at the output.

Plus: when we well drop Symfony < 5, we'll just have to remove the TreeBuilder.php file and add `use Symfony\Component\Config\Definition\Builder\TreeBuilder;` at the top of the file.

Comment (tvbeek) [reply src/main/php/PDepend/DependencyInjection/Configuration.php]: Good point

File: .travis.yml [hunk 2]
```diff
@@ -53,7 +53,7 @@ before_script:
   - phpenv config-rm xdebug.ini || echo "XDebug is not enabled"
   - composer self-update
   - if [ "$DEPENDENCIES" = "low" ]; then composer update --prefer-dist --prefer-lowest --prefer-stable; fi
-  - if [ "$DEPENDENCIES" != "low" ]; then composer update --prefer-dist --prefer-stable; fi
+  - if [ "$DEPENDENCIES" != "low" ]; then composer config minimum-stability beta && composer update --prefer-dist; fi
```
Comment (tvbeek) [inline .travis.yml]: I think you can undo this change now Symfony 5 is released.

Comment (kylekatarnls) [reply .travis.yml]: Done.

**Comment by cawolf** [comment]
Hi, who is responsible for the next step in this PR? We don't want to miss the features of phpmd/phpmd , but we also want to update to symfony 5 asap. :)

**Comment by tvbeek** [comment]
Another maintainer need to approve this PR before it can be merged.

**Comment by kylekatarnls** [comment]
Hi @cawolf, this has still good chances to be in 2.6.0 (will be released in December).

**Comment by garak** [comment]
@kylekatarnls can you approve this? Pretty please 🙏

**Comment by kylekatarnls** [comment]
Hi @garak I can't approve my own PR, maybe @ravage84 could take a look.

**Comment by ravage84** [comment]
Thanks everybody

**Comment by cawolf** [comment]
Hi again,

thanks for your effort, but was there maybe an error when releasing 2.6.0? `composer` updates fail on our side, because in https://github.com/pdepend/pdepend/blob/2.6.0/composer.json , there is still no symfony 5 supported.

    Problem 1
    - pdepend/pdepend 2.6.0 requires symfony/dependency-injection ^2.3.0|^3|^4 -> satisfiable by symfony/dependency-injection[...]

**Comment by tvbeek** [comment]
@cawolf thanks for reporting.

It looks like the 2.6.0 tag was not on the master branch, it is on 713e14e401d174882a5980446db1ba427922310b that is merged with #402 (and that branch didn't had this pr)

@ravage84 and @kylekatarnls I think we need a new release to fix this, or is there a way to restore it?

**Comment by ravage84** [comment]
@tvbeek let's create 2.6.1 for that. Numbers are cheap...

**Comment by kylekatarnls** [comment]
So #424 to be merged first.
